apkid/rules/dex/obfuscators.yara

```diff
@@ -140,3 +140,33 @@ rule arxan : obfuscator
 }
 
 
+rule arxan_multidex : obfuscator
+{
+  meta:
+    description = "Arxan (multidex)"
+    url         = "https://www.arxan.com/products/application-protection-mobile/"
+    example     = "9b2a978a937293d6cb93439e0f819b4e044a3fad80dde92dec9b67e419278b5d"
+
+  strings:
+    // Obfuscated Lpackage/class/: "L([a-z]\1{5}\/[a-z]{6}\/".
+    // AFAIK, Yara does not support backreferences at the moment, thus this is the combo:
+    $pkg = /L(a{6}|b{6}|c{6}|d{6}|e{6}|f{6}|g{6}|h{6}|i{6}|j{6}|k{6}|l{6}|m{6}|n{6}|o{6}|p{6}|q{6}|r{6}|s{6}|t{6}|u{6}|v{6}|w{6}|x{6}|y{6}|z{6})\/[a-z]{6}/
+
+    // Obfuscated methods are found to follow a pattern like:
+    // 1 byte size + 1 byte ASCII + [7-26] non-ASCII bytes + 00 (null terminator)
+    $m1 = { 10 62 (6? | 75) [14] 00 }
+    $m2 = { (0b | 0d) 62 d0 [15] 00 }
+    $m3 = { (0e | 10) 62 30 34 3? [15] 00 }
+    $m4 = { (0b | 0d) 62 30 34 3? [13] 00 }
+    $m5 = { (08 | 0b | 0d | 0e ) 62 [7-13] 00 }
+    $m6 = { 0a 62 (30 34 3? | d? ?? ??) [11] 00 }
+    $m7 = { (0d | 0b | 11) (62 d1 8? | 6? ?? ??) [14] 00 }
+
+  condition:
+    is_dex and
+    $pkg and
+    2 of ($m*) and
+    not arxan
```

Comment:
The way this is written with `not arxan`, makes me think that we may hit a situation where `arxan` fires, however this may be a more correct detection. Since this rule explicitly doesn't file when `arxan` does, would it be better to check in the `arxan` rule that `arxan_mutlidex` doesn't file?

I haven't tested either of these rules, so I'm unsure what the overlap (if any) is.

Reply:
There is no overlap, it is the way that APKiD scans file by file and it may fail the original rule which has 6 out of 7 methods. When multidex happens, the original rule may fail because the code is split in several dex files.